SampleApplications/Samples/XamarinClient/XamarinClient/SampleClient.cs

```diff
@@ -0,0 +1,469 @@
+                    value = rawValue.Replace("|","\r\n").Replace("{","").Replace("}","");
+                }
+                return value;
+            }
+            catch (Exception exception)
+            {
+                return null;
+            }
+        }
+
+        private void CertificateValidator_CertificateValidation(CertificateValidator validator, CertificateValidationEventArgs e)
+        {
+            e.Accept = (e.Error.StatusCode == StatusCodes.BadCertificateUntrusted);
+            // This will just accept any certificate
+            //e.Accept = true;
+        }
+
+        private static EndpointDescriptionCollection DiscoverEndpoints(ApplicationConfiguration config, Uri discoveryUrl, int timeout)
+        {
+            // use a short timeout.
+            EndpointConfiguration configuration = EndpointConfiguration.Create(config);
+            configuration.OperationTimeout = timeout;
+
+            using (DiscoveryClient client = DiscoveryClient.Create(
+                discoveryUrl,
+                EndpointConfiguration.Create(config)))
+            {
+                try
+                {
+                    EndpointDescriptionCollection endpoints = client.GetEndpoints(null);
+                    ReplaceLocalHostWithRemoteHost(endpoints, discoveryUrl);
+                    return endpoints;
+                }
+                catch (Exception e)
+                {
+                    throw e;
+                }
+            }
+        }
+
+        private static EndpointDescription SelectUaTcpEndpoint(EndpointDescriptionCollection endpointCollection, bool haveCert)
+        {
+            EndpointDescription bestEndpoint = null;
+            foreach (EndpointDescription endpoint in endpointCollection)
+            {
+                if (endpoint.TransportProfileUri == Profiles.UaTcpTransport)
+                {
+                    if (bestEndpoint == null ||
+                        haveCert && (endpoint.SecurityLevel > bestEndpoint.SecurityLevel) ||
+                        !haveCert && (endpoint.SecurityLevel < bestEndpoint.SecurityLevel))
+                    {
+                        bestEndpoint = endpoint;
+                    }
+                }
+            }
+            return bestEndpoint;
+        }
+
+        private static void ReplaceLocalHostWithRemoteHost(EndpointDescriptionCollection endpoints, Uri discoveryUrl)
+        {
+            foreach (EndpointDescription endpoint in endpoints)
+            {
+                endpoint.EndpointUrl = Utils.ReplaceLocalhost(endpoint.EndpointUrl, discoveryUrl.DnsSafeHost);
+                StringCollection updatedDiscoveryUrls = new StringCollection();
+                foreach (string url in endpoint.Server.DiscoveryUrls)
+                {
+                    updatedDiscoveryUrls.Add(Utils.ReplaceLocalhost(url, discoveryUrl.DnsSafeHost));
+                }
+                endpoint.Server.DiscoveryUrls = updatedDiscoveryUrls;
+            }
+        }
```

Comment:
should be obsolete? now done in getendpoints

Reply:
yes removed